minimal-plugin/src/main/java/io/micronaut/gradle/MicronautComponentPlugin.java

```diff
@@ -80,7 +80,14 @@ public void apply(Project project) {
         MicronautExtension micronautExtension = project.getExtensions().getByType(MicronautExtension.class);
         TaskContainer tasks = project.getTasks();
         TaskProvider<ApplicationClasspathInspector> inspectRuntimeClasspath = registerInspectRuntimeClasspath(project, tasks);
-
+        TaskProvider<GenerateImportFactoryTask> generateImportFactories = project.getTasks().register(
+                "generateImportFactories",
+                GenerateImportFactoryTask.class,
+                task -> {
+                    task.getRuntimeClasspath().from(project.getConfigurations().getByName(JavaPlugin.RUNTIME_CLASSPATH_CONFIGURATION_NAME));
+                    task.getTaskEnabled().set(true);
+                }
+        );
```

Comment:
I would move this to `configureJava` so that you don't have to search for the created task inside.